src/components/ingestion/AbstractIngestionEngine.js

```diff
@@ -5,61 +5,90 @@ import {
 	AbstractInstantiationError,
 	NotImplementedError,
 } from '@tvkitchen/base-errors'
+import { TSDemuxer } from 'ts-demuxer'
+
 import logger from '%src/lib/logger'
 import kafka from '%src/lib/kafka'
+import { tsToSeconds } from '%src/lib/utils/mpegts'
 
 class AbstractIngestionEngine {
 	constructor() {
 		if (this.constructor === AbstractIngestionEngine) {
 			throw new AbstractInstantiationError(this.constructor.name)
 		}
 		this.producer = kafka.producer()
+		this.mpegtsDemuxer = new TSDemuxer(this.processDemuxedPacket)
+		this.ffmpegProcess = null
+		this.currentPts = 0
 	}
 
 	/**
-	 * Ingests data into the TV Kitchen pipeline.
+	 * Ingests a mpegts payload into the TV Kitchen pipeline.
```

Comment:
`an MPEG-TS payload`…

Reply:
I saw that when we were on-screen this afternoon and smirked at myself.

I swear I know how to use indefinite articles.

Reply:
You indefinitely don't.